stravalib/tests/integration/conftest.py

```diff
@@ -0,0 +1,16 @@
+import pytest
+
+from stravalib import Client
+from stravalib.tests.integration.strava_api_stub import StravaAPIMock
+
+
+@pytest.fixture
+def mock_strava_api():
```

Comment:
@lwasser , I'll jump in and provide a little context around this.  What @jsamoocha is doing -- and honestly, this is brilliant -- is using Strava's published Swagger/OpenAPI JSON (https://developers.strava.com/swagger/swagger.json) to extract all the known API paths and the examples (which are in that document) that Strava provides as a sort of documentation.  To address your question more specifically, the mock here serves to intercept and replace what would be the real HTTP requests out to the Strava API (which would require an application ID / secret / athletes / etc.) with the responses that were provided by Strava as examples in their Swagger.

I particularly like this approach as:
1. As mentioned above, this removes the need for having to set up any real infrastructure (accounts / apps / etc.) to run the tests
2. The example responses are pulled down with every test run.  Since these will be kept up-to-date by Strava we'll know that if stravalib starts breaking that something in the API is no longer as expected.

Reply:
Thanks for the explanation, @hozn, while I was busy adding some documentation to the code!

@lwasser, to address your question regarding passthru: because the fetching of the swagger.json file from Strava is happening within the context of the mocked API, that request would've been mocked as well. So we need to use `add_passthru()` to tell the mocker not to interfere with that request.

Reply:
awesome. @jsamoocha thank you for all of this work and for the explanation about how this is working. i'll defer to any issues @hozn sees in the PR which it sounds like they really appreciate your approach !! so please don't let me slow down the process of merging this. it all looks good to me !